crates/ffi/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "glaredb_ffi"
+version.workspace = true
+edition.workspace = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+arrow = { version = "50.0.0", features = ["ffi"], default-features = false }
+arrow-schema = { version = "50.0.0", features = ["ffi"] }
+thiserror.workspace = true
+
+datafusion = { workspace = true, optional = true }
+paste = "1.0.14"
+
+[features]
+datafusion = ["dep:datafusion"]
```

Comment:
i wasn't sure if we'd need datafusion in any/all of these extensions so i made it opt in. This cuts out ~150 dependencies, reduces build time, build size of `extensions/distance` significantly from `5.6MB` to `3.8MB`.